app/helpers/govuk_form_helper.rb

```diff
@@ -0,0 +1,131 @@
+module GovukFormHelper
+  # = GOVUK Form Tag Helpers
+  #
+  # For usage, components and patterns see:
+  # https://design-system.service.gov.uk/
+
+  # Can be used to wrap multiple or single components.
+  #
+  # Creates a div with the 'govuk-form-group' css class, if the provided object has errors, adds
+  # the 'govuk-form-group--error' css class.
+  #
+  # Passing the attribute symbol will only add the error class if the object contains errors for
+  # the matching symbol.
+  #
+  # Generally pass only an object when wrapping multiple components and pass an object and attribute when
+  # wrapping a single component.
+  #
+  # === Examples
+  # ==== Wrap multiple components
+  #
+  #   <%= form_group_tag object do %>
+  #     <fieldset class="govuk-fieldset">
+  #       <legend class="govuk-fieldset__legend govuk-fieldset__legend--xl">
+  #         <h1 class="govuk-fieldset__heading">
+  #           Where do you live?
+  #         </h1>
+  #       </legend>
+  #     ...
+  #     </fieldset>
+  #   <% end %>
+  #
+  #   <div class="govuk-form-group">
+  #     <fieldset class="govuk-fieldset">
+  #       <legend class="govuk-fieldset__legend govuk-fieldset__legend--xl">
+  #         <h1 class="govuk-fieldset__heading">
+  #           Where do you live?
+  #         </h1>
+  #       </legend>
+  #        ...
+  #     </fieldset>
+  #   </div>
+  #
+  # ==== Wrap a single component
+  #
+  #   <%= form_group_tag object, :event_name do %>
+  #     ...
+  #   <% end %>
+  #
+  #   <div class="govuk-form-group">
+  #     <label class="govuk-label" for="object_event_name">
+  #       Event name
+  #     </label>
+  #     <input class="govuk-input" id="object_event_name" name="event_name" type="text">
+  #   </div>
+  #
+  def form_group_tag(object, attribute = nil, &block)
+    css_classes = ["govuk-form-group"]
+    case attribute
+    when Symbol
+      css_classes << "govuk-form-group--error" if object.errors.key?(attribute)
+    else
+      css_classes << "govuk-form-group--error" if object.errors.any?
+    end
+    content_tag(:div, class: css_classes) do
+      yield
+    end
+  end
+
+  # Returns a set of +<span>+ tags that contains the error messages in _object_ that match the
+  # _attribute_ symbol.
+  #
+  # === Examples
+  #   errors_tag(object, :attribute)
+  #
+  #   <span id="object_attibute_name-error" class="govuk-error-message">
+  #     <span class="govuk-visually-hidden">Error: </span>Error message one<br>
+  #     <span class="govuk-visually-hidden">Error: </span>Error message two
+  #   </span>
+  #
+  def errors_tag(object, attribute)
+    return if object.errors.empty?
+    messages = object.errors.messages[attribute]
+
+    messages.map! do |message|
```

Comment:
Does `!`ing this mean we're modifying `object.errors.messages[attribute]`, since we're interacting with a reference?

Reply:
Yes, I wasn't sure about that, I've got the `dup` now do you think that is better?

Reply:
I think that works. You could also reassign with a plain `map`, e.g. `messages = message.map do ...`

Reply:
Yeah, I prefer your simpler approach.